dsa/dsa.go

```diff
@@ -0,0 +1,309 @@
+// Copyright 2011 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package dsa implements the Digital Signature Algorithm, as defined in FIPS 186-3.
+//
+// The DSA operations in this package are not implemented using constant-time algorithms.
+//
+// Deprecated: DSA is a legacy algorithm, and modern alternatives such as
```

Comment:
I think we'll want to drop the `Deprecated:` bit here to avoid further `staticcheck` results. Maybe use `Warning:` instead?

Reply:
Huh, I admit that I didn't know that [godoc](https://github.com/golang/go/wiki/Deprecated) picks up on that (I've honestly paid more attention to Rust documentation capabilities over the past several years.).

Reply:
It was a TIL for me too. It seems underdocumented! The best source I could find was one small sentence from https://blog.golang.org/godoc

> To signal that an identifier should not be used, add a paragraph to its doc comment that begins with "Deprecated:" followed by some information about the deprecation

Reply:
>  I didn't know that [godoc](https://github.com/golang/go/wiki/Deprecated) picks up on that 

That's a better reference, ty!